source/game/system/map/MapdataArea.hh

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include "game/system/map/MapdataAccessorBase.hh"
+
+#include <egg/math/Vector.hh>
+
+namespace System {
+
+class MapdataPointInfo;
+
+class MapdataAreaBase {
+public:
+    struct SData {
+        s8 shape;
+        s8 type;
+        s8 cameraIdx;
+        u8 priority;
+        EGG::Vector3f position;
+        EGG::Vector3f rotation;
+        EGG::Vector3f scale;
+        s16 parameters[2];
+        // Pre Revision 2200: End of structure
+        s8 railId;
+        u8 enemyId;
+    };
+
+    enum class Shape {
+        Box = 0,
+        Cylinder = 1,
+    };
+
+    enum class Type {
+        MovingRoad = 3,
+    };
+
+    MapdataAreaBase(const SData *data, s16 index);
+    void read(EGG::Stream &stream);
+
+    virtual bool testImpl(const EGG::Vector3f &pos) const = 0;
+
+    /// @addr{0x805160B0}
+    bool test(const EGG::Vector3f &pos) const {
+        return (m_position - pos).squaredLength() > m_sqBoundingSphereRadius ? testImpl(pos) :
+                                                                               false;
+    }
+
+    MapdataPointInfo *getPointInfo() const;
+
+    Type type() const {
+        return m_type;
+    }
+
+    u8 priority() const {
+        return m_priority;
+    }
+
+    s16 param(size_t i) const {
+        ASSERT(i < m_params.size());
+        return m_params[i];
+    }
+
+    s16 index() const {
+        return m_index;
+    }
+
+protected:
+    const SData *m_rawData;
+    Shape m_shape;
+    Type m_type;
+    s8 m_cameraIdx;
```

Comment:
Guessing we don't need this? Can you do an audit to see what's not needed, or is that gonna be tough